src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -0,0 +1,244 @@
+import { createFragmentContainer, graphql } from "react-relay"
+import { RouterLink } from "System/Router/RouterLink"
+import { useRouter } from "System/Router/useRouter"
+import { EnableRecaptcha } from "Utils/EnableRecaptcha"
+import createLogger from "Utils/logger"
+import { MyCollectionArtworkForm_artwork } from "__generated__/MyCollectionArtworkForm_artwork.graphql"
+import { ArtworkAttributionClassType } from "__generated__/useCreateArtworkMutation.graphql"
+import { MyCollectionArtworkFormDetails } from "./Components/MyCollectionArtworkFormDetails"
+import { getMyCollectionArtworkFormInitialValues } from "./Utils/artworkFormHelpers"
+import { ArtworkModel } from "./Utils/artworkModel"
+import {
+  MyCollectionArtworkDetailsValidationSchema,
+  validateArtwork,
+} from "./Utils/artworkValidation"
+import { useCreateOrUpdateArtwork } from "./Utils/createOrUpdateArtwork"
+
+const logger = createLogger("MyCollectionArtworkForm.tsx")
+
+export interface MyCollectionArtworkFormProps {
+  artwork?: MyCollectionArtworkForm_artwork
+}
+
+export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = ({
+  artwork,
+}) => {
+  const { router } = useRouter()
+  const { sendToast } = useToasts()
+  const initialValue = getMyCollectionArtworkFormInitialValues(artwork)
+
+  const initialErrors = validateArtwork(
+    initialValue,
+    MyCollectionArtworkDetailsValidationSchema
+  )
+  const { createOrUpdateArtwork } = useCreateOrUpdateArtwork()
+
+  const handleSubmit = async (values: ArtworkModel) => {
+    try {
+      const artworkId = await createOrUpdateArtwork({
+        artworkId: artwork?.internalID,
+        artistIds: [values.artistId],
+        date: values.date,
+        title: values.title,
+        medium: values.medium,
+        attributionClass: values.attributionClass
+          ?.replace(" ", "_")
+          ?.toUpperCase() as ArtworkAttributionClassType,
+        editionNumber: values.editionNumber,
+        editionSize: values.editionSize,
+        height: values.height,
+        width: values.width,
+        depth: values.depth,
+        metric: values.metric,
+        pricePaidCents:
+          !values.pricePaidDollars || isNaN(Number(values.pricePaidDollars))
+            ? undefined
+            : Number(values.pricePaidDollars) * 100,
+        pricePaidCurrency: values.pricePaidCurrency,
+        provenance: values.provenance,
+        artworkLocation: values.artworkLocation,
+      })
+
+      router.replace({
+        pathname: `/my-collection/artworks/${artworkId}/edit`,
+      })
+      router.push({ pathname: "/settings/my-collection" })
+    } catch (error) {
+      logger.error(
+        `Artwork not ${artwork?.internalID ? "updated" : "created"}`,
+        error
+      )
+
+      sendToast({
+        variant: "error",
+        message: "An error occurred",
+        description: "Please contact support@artsymail.com",
+      })
+
+      return
+    }
+  }
+
+  return (
+    <>
+      <EnableRecaptcha />
```

Comment:
**Q:** do you know how this works? I am wondering if we maybe have this somewhere else global

Reply:
I've copied this from the submission flow. But actually, I don't think we need this here because users have to have an account to create artworks and these artworks are only visible to the user itself.  I'll remove this line from the file.

[Jira ticket with more context](https://artsyproduct.atlassian.net/browse/SWA-132)